packages/theme-language-server-common/src/docset/MarkdownRenderer.ts

```diff
@@ -67,6 +68,22 @@ function htmlEntryBody(entry: HtmlEntry, parentEntry?: HtmlEntry) {
     .join(HORIZONTAL_SEPARATOR);
 }
 
+function syntax(entry: DocsetEntry | FilterEntry | TagEntry) {
+  if (!('syntax' in entry) || !entry.syntax) {
```

Comment:
This line could be a little confusing since it looks like I'm doing the same check twice but it's a little more nuanced than that! The first check of `!('syntax' in entry)` gives us type narrowing. Only `FilterEntry` and `TagEntry` have the `syntax` attribute but this could be a `DocsetEntry`. If only checked for `entry.syntax` TypeScript would complain because that attribute doesn't exist on `DocsetEntry`. Once we've narrowed the type *then* we can check `!entry.syntax` (the attribute could be there but `undefined`).

Reply:
Ah that's why you need FilterEntry | TagEntry. Because the props are _extra_.